ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/context/KsqlImpersonationHandler.java

```diff
@@ -15,29 +15,21 @@
 
 package io.confluent.ksql.rest.server.context;
 
-import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.rest.server.KsqlRestConfig;
 import io.confluent.ksql.util.KsqlConfig;
 import org.glassfish.hk2.utilities.binding.AbstractBinder;
-import org.glassfish.jersey.process.internal.RequestScoped;
 
 /**
- * Configures the {@link ServiceContext} class for dependency injection using the
- * {@link javax.ws.rs.core.Context} annotation.
+ * Abstract class to handle KSQL user impersonation. Subclass implementation needs to implement
+ * the {@link AbstractBinder} methods and configure the binder.
  * </p>
- * Inject {@code ServiceContext} on each REST method as follows:
- * i.e. myMethod(@Context ServiceContext serviceContext)
+ * The subclass implementation must be registered with the
+ * {@link KsqlRestConfig#KSQL_IMPERSONATION_HANDLER_CLASS} configuration.
  */
-public class KsqlRestServiceContextBinder extends AbstractBinder {
-  private final KsqlConfig ksqlConfig;
+public abstract class KsqlImpersonationHandler extends AbstractBinder {
```

Comment:
I think including `Impersonation` in the name of this class is misleading. You can do whatever configuration you want within it, independent of impersonation. Instead I would call it `ServiceContextBinder` and rename `KsqlDisableImpersonationHandler` to `DefaultServiceContextBinder` (and same applies to everywhere the word "impersonation" is used)